networking/macsec/sanity_check/runtest.sh

```diff
@@ -0,0 +1,506 @@
+        {
+            head /dev/urandom | md5sum | awk '{print $1}'
+        }
+        rlRun "ip macsec add macsec0 tx sa 2 pn 100 key 03 123456789123456789" 2 "bad key length"
+        hex32_01=$(gen_random_hex32)
+        hex32_02=$(gen_random_hex32)
+        rlRun "ip macsec add macsec0 tx sa 2 pn 100 key $hex32_01 $hex32_02"
+        rlRun "ip macsec show | grep $hex32_01"
+
+        # port number
+        rlRun "ip macsec set macsec0 tx sa 2 pn 1024 off"
+        rlRun "ip macsec set macsec0 tx sa 2 pn 1024 on"
+
+        # del
+        rlRun "ip macsec del macsec0 rx sci $SCI_PEER"
+        # Bug 1357349 - MACsec: fail to delete macsec tx/rx sa
+
+    rlPhaseEnd
+
+    rlPhaseStartTest "check MTU"
+        # the SecTAG before Secure Data is 16 tytes
+        # the ICV after Secure Data is 16 bytes by default, which can be set by icvlen (8..16)
+        modprobe -r macsec
+        modprobe -r dummy
+        if [ $(GetDistroRelease) -ge 8 ]; then
+                modprobe dummy numdummies=1
+        else
+                modprobe dummy
+        fi
+
+        rlRun "ip link add link dummy0 macsec0 type macsec"
+        rlAssertEquals "default real_dev MTU is 1500" $(cat /sys/class/net/dummy0/mtu) 1500
+        rlAssertEquals "default macsec header + tailer is 32" $(cat /sys/class/net/macsec0/mtu) $((1500-32))
+        rlRun "ip link del macsec0"
+        rlRun "ip link add link dummy0 macsec0 type macsec cipher default icvlen 8"
+        rlAssertEquals "set macsec tailer to 8" $(cat /sys/class/net/macsec0/mtu) $((1500-16-8))
+        rlRun "ip link del macsec0"
+        rlRun "ip link add link dummy0 macsec0 type macsec cipher default icvlen 16"
+        rlAssertEquals "set macsec tailer to 16" $(cat /sys/class/net/macsec0/mtu) $((1500-16-16))
+        rlRun "ip link del macsec0"
+
+        # macsec MTU value range should less than real_dev's
+        rlRun "ip link add link dummy0 macsec0 type macsec"
+        rlRun "ip link set dummy0 mtu 9000"
+        rlRun "ip link set macsec0 mtu 9000" 2
+        rlRun "ip link set macsec0 mtu 100"
+        rlRun "ip link set macsec0 mtu 1500"
+        rlRun "ip link set macsec0 mtu 3000"
+        rlRun "ip link set macsec0 mtu 8968"
+
+        # macsec MTU value should be reduced automatiaclly to adapt to real_dev
+        for i in 9000 5000 1500 100; do
+            sleep 1
+            rlRun "ip link set dummy0 mtu $i"
+            rlAssertEquals "macsec should adapt to real_dev" $(cat /sys/class/net/macsec0/mtu) $((i-32))
+        done
+        rlRun "ip link del macsec0"
+    rlPhaseEnd
+
+    rlPhaseStartTest "check promiscuous mode"
+        function if_dev_promisc()
+        {
+            # return 1 for in promisc mode
+            # return 0 for in nopromisc mode
+            local dev=$1
+            local dev_flags=$(cat /sys/class/net/$dev/flags)
+            dev_promisc_info="$dev: $dev_flags"
+            python -c "import sys; ret=int(($dev_flags & 0x100)>>8); exit(ret)"
```

Comment:
Do we need to set which version of python we're using here ?

Reply:
`(( cmd ))` is able to do the same operation. 
```patch
-            python -c "import sys; ret=int(($dev_flags & 0x100)>>8); exit(ret)"
+            local ret=$(((dev_flags & 0x100) >> 8))
+            return $ret
```